java/java.hints/src/org/netbeans/modules/java/hints/infrastructure/JavaErrorProvider.java

```diff
@@ -94,7 +95,9 @@ public class JavaErrorProvider implements ErrorProvider {
     @Override
     public List<? extends Diagnostic> computeErrors(Context context) {
         List<Diagnostic> result = new ArrayList<>();
-
+        if (!GraphicsEnvironment.isHeadless()) {
```

Comment:
The `java.lsp.server` project is using `isHeadless()` to detect whether the code is running in VSCode or in NetBeans IDE. Using it here to solve https://github.com/apache/netbeans/pull/7579#issuecomment-2236478654 inquiry.

Reply:
This gives me bad vibes and is only a partial solution. I placed a breakpoint in `ErrorProviderBridge#run` and I see three `ErrorProvider` there:

1. `org.netbeans.modules.java.hints.infrastructure.JavaErrorProvider`
2. `org.netbeans.modules.editor.hints.lsp.HintsDiagnosticsProvider`
3. `org.netbeans.modules.micronaut.symbol.MicronautSymbolErrorProvider`

(1) is now handled by checking `GraphicsEnvironment#isHeadless`, (2) ist explicitly filtered out by its name, (3) is retained.

This will become whac-a-mole because the interaction are not obvious. Would a different approach make sense where an autoloaded helper module, that depends on the modules, that provides the LSP implementation, provides the bridge only for that module? That way the interaction is easier to see and if necessary easy to drop. 

I don't know the enso module name but assuming it is `enso-lsp`, a module `enso-lsp.nb` could be used to provide the integration limited to the enso type.